tests/unit/test_polynomial_decay_derivative.py

```diff
@@ -0,0 +1,28 @@
+import numpy as np
+import pytest
+
+from gradient_aware_harmonisation.convergence import (
+    PolynomialDecaySplineHelperDerivative,
+)
+
+scipy = pytest.importorskip("scipy")
+
+
+def test_PolynomialDecaySplineHelperDerivative():
```

Comment:
```suggestion
def test_polynomial_decay_spline_helper_derivative():
```

(nwtkia) For whatever reason, the convention is to use snake_case for function names, even if we're directly referring to a class name